packages/suite/src/middlewares/wallet/storageMiddleware.ts

```diff
@@ -311,6 +311,8 @@ const storageMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => {
                 case FORM_DRAFT.REMOVE_DRAFT:
                     storageActions.removeFormDraft(action.key);
                     break;
+
+                case deviceActions.connectDevice.type: // so that firmwareReducer.addCase for the same action is persisted
```

Comment:
this works because `next(action)` is called [at beginning](https://github.com/trezor/trezor-suite/blob/509ff96ac221542d9c1f20b54b2f370f8bbaf1d7/packages/suite/src/middlewares/wallet/storageMiddleware.ts#L46).
So the `firmwareReducer.addCase` does its thing first, and only then this `case` block is executed to persist the newly updated `state.firmware.firmwareHashInvalid`.